.emacs.d/config.org

```diff
@@ -55,62 +55,392 @@
+    :custom
+    (projectile-project-search-path '("~/Projects/"))
+    (projectile-enable-caching t)
+    (projectile-require-project-root t)
+    (projectile-sort-order 'access-time)
+    :init
+    (projectile-mode))
+  #+end_src
+
+* Magit
+
+  [[https://github.com/magit/magit][Magit]] is a Git interface, just like you have in many IDEs and text editors, but combining that with all the power of using Git from the command line.
+
+  #+begin_src emacs-lisp
+  (use-package magit)
+  #+end_src
+
+* Key bindings
+
+  [[https://github.com/justbur/emacs-which-key][Which-key]] is a package that displays what bindings are available when you start pressing a key. It is incredibly useful, especially when you are new to Emacs or when you are trying a new package.
+
+  #+BEGIN_SRC emacs-lisp
+  (use-package which-key
+    :init
+    (which-key-mode))
+  #+END_SRC
+
+* Linters
+
+  The two main frameworks that allow Emacs to interface with external linters are
+  - [[https://www.gnu.org/software/emacs/manual/html_node/emacs/Flymake.html][flymake]]
+  - [[https://www.flycheck.org/en/latest/][flycheck]]
+  A detailed comparison is at https://www.flycheck.org/en/latest/user/flycheck-versus-flymake.html. We choose to use flycheck due to the sheer number of extensions written for it; pick your favorite language linter, and there’s probably a flycheck extension for it.
+
+  #+BEGIN_SRC emacs-lisp
+  (use-package flycheck)
+  #+END_SRC
+
+* Language and debug servers
+
+  [[https://microsoft.github.io/language-server-protocol][LSP]] is a protocol used to implement IDE-like features in an editor-agnostic manner. This package will give us a uniform way to add features such as semantic highlighting, auto-completion, jump-to-definition, and so on.
+
+  #+BEGIN_SRC emacs-lisp
+  (use-package lsp-mode
+    :commands lsp
+    :hook ((c-mode c++-mode python-mode sh-mode) . lsp)
+    :custom
+    (lsp-auto-guess-root t)
+    (lsp-log-io t)
+    (lsp-prefer-flymake nil))
+  #+END_SRC
+
+  This package enhances the user interface, providing pop-up windows with documentation and visual indicators.
+
+  #+BEGIN_SRC emacs-lisp
+  (use-package lsp-ui
+    :hook (lsp-mode . lsp-ui-mode)
+    :custom
+    (lsp-ui-doc-delay 3.0)
+    (lsp-ui-doc-include-signature t)
+    (lsp-ui-flycheck-enable t)
+    (lsp-ui-peek-always-show t))
+  #+END_SRC
+
+  Analogous to LSP is the [[https://microsoft.github.io/debug-adapter-protocol/][debug adapter protocol]]. We use [[https://github.com/emacs-lsp/lsp-treemacs][treemacs]] for IDE-like display of errors.
+
+  #+BEGIN_SRC emacs-lisp
+  (use-package dap-mode
+    :after lsp-mode
+    :config
+    (dap-mode t)
+    (dap-ui-mode t))
+
+  (use-package treemacs
+    :bind (:map global-map
+                ("C-x t t" . treemacs)
+                ("C-x t 1" . treemacs-select-window))
+    :custom
+    (treemacs-resize-icons 15))
+
+  (use-package lsp-treemacs
+    :init (lsp-treemacs-sync-mode 1)
+
+    :bind (:map java-mode-map
```

Comment:
This should spit out some errors as java map isn't defined yet or something along those lines...

Reply:
The behavior I'm getting is that those simply aren't bound outside of `java-mode`, so it works fine for me. Try it and confirm?